### ISSUE TITLE
Latest Bundler (2.1.4) fails to install on TruffleRuby

### ISSUE THREAD

**Issue by eregon** [oracle/truffleruby#1880]
Using < /dev/null to simulate non-interactive shell:
```
$ gem install bundler < /dev/null 
ERROR:  Error installing bundler:
	"bundle" from bundler conflicts with /home/eregon/.rubies/truffleruby-19.3.0/bin/bundle
```

Also seen when running `gem install bundler` in GitHub Actions.

When run with an interactive shell it asks:
```
$ gem install bundler
Fetching bundler-2.1.4.gem
bundler's executable "bundle" conflicts with /home/eregon/.rubies/truffleruby-19.3.0/bin/bundle
Overwrite the executable? [yN]  
```

On MRI, it installs without asking anything:
```
Fetching bundler-2.1.4.gem
Successfully installed bundler-2.1.4
1 gem installed
```

I think this is because our `bin/bundle` is not the traditional RubyGems stub but something else.
We should do the same as MRI here most likely.

**Comment by bjfish** [comment]
@eregon FWIW: I was able to `gem update bundler` with native, I have a GEM_HOME set, and using rbenv.

**Comment by eregon** [comment]
Setting `GEM_HOME` seems indeed to workaround the problem, since then the executable ends up in `$GEM_HOME/bin/bundle`:
```
GEM_HOME=/.../gem_home gem install bundler
Fetching bundler-2.1.4.gem
Successfully installed bundler-2.1.4
1 gem installed
```

**Comment by bjfish** [comment]
It appears this section detects the existing stub and early returns:
https://github.com/ruby/ruby/blob/ruby_2_6/lib/rubygems/installer.rb#L231-L241
It is able to get the name from this section:
https://github.com/ruby/ruby/blob/650c45dcd70e791747c4b9084c100cb895e718cb/bin/bundle#L22-L27

**Comment by eregon** [comment]
Indeed, that seems likely to be the logic causing this.
Currently we don't match that Regexp:
https://github.com/oracle/truffleruby/blob/687277be74e391502cafc4d61ef2712a9399f9f4/bin/bundle

MRI seems to have the non-stub `bundle` executable under `libexec`:
https://github.com/ruby/ruby/blob/v2_6_5/libexec/bundle

**Comment by eregon** [comment]
Fixed in 49f199abd48fbc51f26e0af8f38933c3c8ff67c3 by @bjfish, it will be in the next release.
